sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchAsyncClientTest.java

```diff
@@ -0,0 +1,722 @@
+package com.azure.maps.search;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpClient;
+import com.azure.core.models.GeoLineString;
+import com.azure.core.models.GeoObject;
+import com.azure.core.util.Context;
+import com.azure.core.util.polling.PollerFlux;
+import com.azure.core.util.polling.SyncPoller;
+import com.azure.maps.search.models.BatchReverseSearchResult;
+import com.azure.maps.search.models.BatchSearchResult;
+import com.azure.maps.search.models.FuzzySearchOptions;
+import com.azure.maps.search.models.LatLong;
+import com.azure.maps.search.models.ReverseSearchAddressOptions;
+import com.azure.maps.search.models.ReverseSearchCrossStreetAddressOptions;
+import com.azure.maps.search.models.SearchAddressOptions;
+import com.azure.maps.search.models.SearchAlongRouteOptions;
+import com.azure.maps.search.models.SearchInsideGeometryOptions;
+import com.azure.maps.search.models.SearchNearbyPointsOfInterestOptions;
+import com.azure.maps.search.models.SearchPointOfInterestCategoryOptions;
+import com.azure.maps.search.models.SearchPointOfInterestOptions;
+import com.azure.maps.search.models.SearchStructuredAddressOptions;
+import com.azure.maps.search.models.StructuredAddress;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import reactor.test.StepVerifier;
+
+public class MapsSearchAsyncClientTest extends MapsSearchClientTestBase {
+    private MapsSearchAsyncClient client;
+    private static final String DISPLAY_NAME_WITH_ARGUMENTS = "{displayName} with [{arguments}]";
+
+    @BeforeAll
+    public static void beforeAll() {
+        StepVerifier.setDefaultTimeout(Duration.ofSeconds(30));
+    }
+
+    @AfterAll
+    public static void afterAll() {
+        StepVerifier.resetDefaultTimeout();
+    }
+
+    private MapsSearchAsyncClient getMapsSearchAsyncClient(HttpClient httpClient,
+        MapsSearchServiceVersion serviceVersion) {
+        return getMapsSearchAsyncClientBuilder(httpClient, serviceVersion).buildAsyncClient();
+    }
+
+    // Test async get polygons
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.maps.search.TestUtils#getTestParameters")
+    public void testAsyncGetMultiPolygons(HttpClient httpClient, MapsSearchServiceVersion serviceVersion) {
+        client = getMapsSearchAsyncClient(httpClient, serviceVersion);
+        List<String> geometryIds = Arrays.asList("8bceafe8-3d98-4445-b29b-fd81d3e9adf5", "00005858-5800-1200-0000-0000773694ca");
+        StepVerifier.create(client.getPolygons(geometryIds)).assertNext(actualResults -> 
+        {
+            try {
```

Comment:
why catch the exception here? Won't this cause a false positive? It's preferable to use some higher level constructs to deal with exceptions (expectErrorMatches if testing FOR an exception) or simply let it break, no catching